src/model.rs

```diff
@@ -1819,19 +1830,299 @@ mod tests {
+                .calendar("c3", "2020-01-03")
+                .calendar("c3", "2020-01-04")
+                .vj("VJ:1", |vj| {
+                    vj.block_id("block_1")
+                        .calendar("c1")
+                        .st("SP1", "10:00:00", "10:01:00")
+                        .st("SP2", "11:00:00", "11:01:00");
+                })
+                .vj("VJ:2", |vj| {
+                    vj.block_id("block_1")
+                        .calendar("c2")
+                        .st("SP3", "12:00:00", "12:01:00")
+                        .st("SP4", "13:00:00", "13:01:00");
+                })
+                .vj("VJ:3", |vj| {
+                    vj.block_id("block_1")
+                        .calendar("c3")
+                        .st("SP3", "12:30:00", "12:31:00")
+                        .st("SP4", "13:30:00", "13:31:00");
+                })
+                .build();
+
+            let vj1 = model.vehicle_journeys.get("VJ:1").unwrap();
+            let stop_time = &vj1.stop_times[0];
+            assert_eq!(0, stop_time.pickup_type);
+            assert_eq!(1, stop_time.drop_off_type);
+            let stop_time = &vj1.stop_times.last().unwrap();
+            assert_eq!(0, stop_time.pickup_type); // pickup should be possible since the traveler can stay-in the vehicle
+            assert_eq!(0, stop_time.drop_off_type);
+            let vj2 = model.vehicle_journeys.get("VJ:2").unwrap();
+            let stop_time = &vj2.stop_times[0];
+            assert_eq!(0, stop_time.pickup_type);
+            assert_eq!(0, stop_time.drop_off_type); // drop off on first stop possible if anyone took the stay-in
+            let stop_time = &vj2.stop_times.last().unwrap();
+            assert_eq!(1, stop_time.pickup_type); // impossible to pickup on last stop
+            assert_eq!(0, stop_time.drop_off_type);
+            let vj3 = model.vehicle_journeys.get("VJ:3").unwrap();
+            let stop_time = &vj3.stop_times[0];
+            assert_eq!(0, stop_time.pickup_type);
+            assert_eq!(0, stop_time.drop_off_type); // drop off on first stop possible if anyone took the stay-in
+            let stop_time = &vj3.stop_times.last().unwrap();
+            assert_eq!(1, stop_time.pickup_type);
+            assert_eq!(0, stop_time.drop_off_type);
+        }
+
+        #[test]
+        fn block_id_on_overlapping_calendar_forbidden_pickup() {
+            // like the example 4 but on less days
+            // working days:
+            // days: 01 02 03 04
+            // VJ:1   X  X  X  X
+            // VJ:2   X  X  X
+            // VJ:3            X
+            // VJ:1 has a forbidden pick up at the 2nd stop-time that should be kept
+            let model = transit_model_builder::ModelBuilder::default()
+                .calendar("c1", "2020-01-01")
+                .calendar("c1", "2020-01-02")
+                .calendar("c1", "2020-01-03")
+                .calendar("c1", "2020-01-04")
```

Comment:
Just an idea (resolve comment if you're not :star_struck:): it could be nice to handle iterable to do
```suggestion
                .calendar("c1", ["2020-01-01", "2020-01-02", "2020-01-03", "2020-01-04"])
```